app/client/components/ColumnTransform.ts

```diff
@@ -229,7 +232,7 @@ export class ColumnTransform extends Disposable {
         // Those options are supposed to be set by prepTransformColInfo(TypeTransform) and
         // adjusted by client.
         // TODO: is this really needed? Aren't those options already in the data-engine?
-        JSON.stringify(this._fieldBuilder.options.peek()),
+        JSON.stringify({...this.origWidgetOptions as object, ...this._fieldBuilder.options.peek()}),
```

Comment:
Can you first check that origWidgetOptions is an object? 99% of the time, it is, but this is not a rule (hence the unknow type). Some tests use strings or arrays here.